cardinal/config.go

```diff
@@ -37,89 +31,78 @@ var (
 	}
 
 	defaultConfig = WorldConfig{
+		CardinalNamespace:         DefaultCardinalNamespace,
+		CardinalRollupEnabled:     false,
+		CardinalLogPretty:         false,
+		CardinalLogLevel:          DefaultCardinalLogLevel,
 		RedisAddress:              DefaultRedisAddress,
 		RedisPassword:             "",
-		CardinalNamespace:         DefaultNamespace,
-		CardinalMode:              DefaultRunMode,
-		BaseShardSequencerAddress: "",
-		CardinalLogLevel:          DefaultLogLevel,
-		StatsdAddress:             DefaultStatsdAddress,
-		TraceAddress:              "",
+		BaseShardSequencerAddress: DefaultBaseShardSequencerAddress,
+		BaseShardRouterKey:        "",
+		TelemetryEnabled:          false,
+		TelemetryStatsdAddress:    "",
+		TelemetryTraceAddress:     "",
 	}
 )
 
-type RunMode string
-
 type WorldConfig struct {
-	// Cardinal
-	CardinalMode      RunMode `config:"CARDINAL_MODE"`
-	CardinalNamespace string  `config:"CARDINAL_NAMESPACE"`
-	CardinalLogLevel  string  `config:"CARDINAL_LOG_LEVEL"`
+	// CardinalNamespace The shard namespace for Cardinal. This needs to be unique to prevent signature replay attacks.
+	CardinalNamespace string `config:"CARDINAL_NAMESPACE"`
+
+	// CardinalRollupEnabled When true, Cardinal will sequence and recover to/from base shard.
+	CardinalRollupEnabled bool `config:"CARDINAL_ROLLUP_ENABLED"`
 
-	// Redis
-	RedisAddress  string `config:"REDIS_ADDRESS"`
+	// CardinalLogLevel Determines the log level for Cardinal.
+	CardinalLogLevel string `config:"CARDINAL_LOG_LEVEL"`
+
+	// CardinalLogPretty Pretty logging, disable by default due to performance impact.
+	CardinalLogPretty bool `config:"CARDINAL_LOG_PRETTY"`
+
+	// RedisAddress The address of the redis server, supports unix sockets.
+	RedisAddress string `config:"REDIS_ADDRESS"`
+
+	// RedisPassword The password for the redis server. Make sure to use a password in production.
 	RedisPassword string `config:"REDIS_PASSWORD"`
 
-	// Shard networking
+	// BaseShardSequencerAddress This is the address that Cardinal will use to sequence and recover to/from base shard.
 	BaseShardSequencerAddress string `config:"BASE_SHARD_SEQUENCER_ADDRESS"`
 
-	// Telemetry
-	StatsdAddress string `config:"STATSD_ADDRESS"`
-	TraceAddress  string `config:"TRACE_ADDRESS"`
-	// RouterKey is a token used to secure communications between the game shard and the base shard.
-	RouterKey string `config:"ROUTER_KEY"`
+	// BaseShardRouterKey is a token used to secure communications between the game shard and the base shard.
+	BaseShardRouterKey string `config:"BASE_SHARD_ROUTER_KEY"`
```

Comment:
there are a few more changes needed if you want to change this value. docker compose file(s), changing the os.Getenv calls that expect `ROUTER_KEY`, as well as the world-cli/starter template.

Reply:
Already modified the docker compose in the World Engine repo. Need to wait for this to be merged and a new version is tagged for me to bump the starter game template

Reply:
there is still another `ROUTER_KEY` instance in the docker-compose.yml as well as some os.Getenv calls in EVM

Reply:
done